text/0000-lifecycle-multicall-binary-for-build.md

```diff
@@ -0,0 +1,55 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle as a multicall binary for build phases
+- Start Date: 01/21/2020
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+We should compile the lifecycle as a multi call binary for the build phases of detect, analyze, restore, build, export and rebase while retaining the interface to the phases.
+
+
+# Motivation
+[motivation]: #motivation
+
+- The size of the lifecycle release decreases from 24M to 7M.
+- This should speed up time take to build the lifecycle as far fewer binaries get built.
+- The size of the builders will decrease.
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+We are proposing changing the lifecycle release to contain only the following binaries:
+- build binary: The build phases (detector, analyzer, restorer, builder, exporter, rebaser) of the lifecycle will exist in a single multicall binary
```

Comment:
we should explicitly call whether the "real" binary bears the name of one of the existing phases, or is packaged with a different name (in the existing lifecycle PR I believe it has the name `detector`)